samples/gen-go/client/circuit_logger.go

```diff
@@ -0,0 +1,239 @@
+package client
+
+import (
+	"time"
+
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/afex/hystrix-go/hystrix/metric_collector"
+	"github.com/afex/hystrix-go/hystrix/rolling"
+	"gopkg.in/Clever/kayvee-go.v5/logger"
+)
+
+// LoggingCollector fulfills hystrix's metricCollector interface allowing
+// users to log circuit error rates.
+// It is a slightly modified version of hystrix's built in `DefaultMetricCollector`,
+// It borrows code from `metricExchange` for computing error rates.
+// It also borrows from hystrix's SSE `publishMetrics`.
+type LoggingCollector struct {
+	numRequests *rolling.Number
+	errors      *rolling.Number
+
+	successes     *rolling.Number
+	failures      *rolling.Number
+	rejects       *rolling.Number
+	shortCircuits *rolling.Number
+	timeouts      *rolling.Number
+
+	fallbackSuccesses *rolling.Number
+	fallbackFailures  *rolling.Number
+	totalDuration     *rolling.Timing
+	runDuration       *rolling.Timing
+}
+
+// NewLoggingCollector creates a logging collector for a specific circuit.
+func NewLoggingCollector(name string) metricCollector.MetricCollector {
+	m := &LoggingCollector{}
+	m.Reset()
+	logger := logger.New("wag")
+	go func() {
+		for _ = range time.Tick(5 * time.Second) {
+			cb, _, err := hystrix.GetCircuit(name)
+			if err != nil {
+				continue
+			}
+			now := time.Now()
+			data := map[string]interface{}{
+				"reqCount":                       m.Requests().Sum(now),
+				"errCount":                       m.Errors().Sum(now),
+				"errPct":                         m.ErrorPercent(now),
+				"circuitBreakerOpen":             cb.IsOpen(),
+				"rollingCountSuccess":            uint32(m.Successes().Sum(now)),
+				"rollingCountFailure":            uint32(m.Failures().Sum(now)),
+				"rollingCountThreadPoolRejected": uint32(m.Rejects().Sum(now)),
+				"rollingCountShortCircuited":     uint32(m.ShortCircuits().Sum(now)),
+				"rollingCountTimeout":            uint32(m.Timeouts().Sum(now)),
+				"rollingCountFallbackSuccess":    uint32(m.FallbackSuccesses().Sum(now)),
+				"rollingCountFallbackFailure":    uint32(m.FallbackFailures().Sum(now)),
+				"latencyTotal":                   generateLatencyTimings(m.TotalDuration()),
+				"latencyTotalMean":               m.TotalDuration().Mean(),
+				"latencyExecute":                 generateLatencyTimings(m.RunDuration()),
+				"latencyExecuteMean":             m.RunDuration().Mean(),
```

Comment:
I'm realizing now that I don't log the number of concurrent requests. After looking at the code for that I think the only way to do that is via listening via HTTP to the SSE server...